src/core/printing.c

```diff
@@ -79,6 +79,23 @@ int igraph_real_printf(igraph_real_t val) {
         return printf("%g", val);
     }
 }
+
+int igraph_real_printf_aligned(igraph_real_t val) {
+    if (igraph_finite(val)) {
+        return printf("%8g", val);
+    } else if (igraph_is_nan(val)) {
+        return printf("     NaN");
+    } else if (igraph_is_inf(val)) {
+        if (val < 0) {
+            return printf("    -Inf");
+        } else {
+            return printf("     Inf");
+        }
+    } else {
```

Comment:
But this cannot possibly happen, right? If so, it's better to put an `IGRAPH_ASSERT` or `IGRAPH_FATAL` here so we become aware of any problems instead of blindly trying to work around them, without even knowing if they exist.

Reply:
Yes, I just copied it, but it should just give an error.